<?xml version="1.0" ?>


<robot name="StereoVisionSystem"> 

    <material name="white">
        <color rgba="1 1 1 1"/>
    </material>

    <material name="blue">
        <color rgba="0 0 0.8 1"/>
    </material>

    <material name="green">
        <color rgba="0 0.8 0 1"/>
    </material>


    <link name="base_link">
    </link>
<!-- fisheye cameras -->
    <link name="fl_fy" >
        <visual>
        <geometry>
            <sphere radius="0.01"/>
        </geometry>
        <material name="white"/>
        </visual>
    </link>
    <link name="fr_fy" >
        <visual>
        <geometry>
            <sphere radius="0.01"/>
        </geometry>
        <material name="white"/>
        </visual>
    </link>
    <link name="bl_fy" >
        <visual>
        <geometry>
            <sphere radius="0.01"/>
        </geometry>
        <material name="white"/>
        </visual>
    </link>
    <link name="br_fy" >
        <visual>
        <geometry>
            <sphere radius="0.01"/>
        </geometry>
            <material name="white"/>
        </visual>
    </link>
<!-- fisheye joints -->
    <joint name ="base_to_fl" type="fixed">
        <parent link="base_link" />
        <child link="fl_fy"/>
        <origin xyz="-0.1 0.1 0.0" rpy="-1.57 0 0.785"/>
    </joint>
    <joint name ="base_to_fr" type="fixed">
        <parent link="base_link" />
        <child link="fr_fy"/>
        <origin xyz="0.1 0.1 0.0" rpy="-1.57 0 -0.785"/>
    </joint>
    <joint name ="base_to_bl" type="fixed">
        <parent link="base_link" />
        <child link="bl_fy"/>
        <origin xyz="-0.1 -0.1 0.0" rpy="-1.57 0 -3.925"/>
    </joint>
    <joint name ="base_to_br" type="fixed">
        <parent link="base_link" />
        <child link="br_fy"/>
        <origin xyz="0.1 -0.1 0.0" rpy="-1.57 0 3.925"/>
    </joint>
<!-- pinhole cameras -->
    <link name="fl_l_ph" >
        <visual>
        <geometry>
            <box size="0.005 0.02 0.02"/>
        </geometry>
        <material name="blue"/>
        </visual>
    </link>
    <link name="fl_r_ph" >
        <visual>
        <geometry>
            <box size="0.005 0.02 0.02"/>
        </geometry>
        <material name="green"/>
        </visual>
    </link>

    <link name="fr_l_ph" >
        <visual>
        <geometry>
            <box size="0.005 0.02 0.02"/>
        </geometry>
        <material name="blue"/>
        </visual>
    </link>
    <link name="fr_r_ph" >
        <visual>
        <geometry>
            <box size="0.005 0.02 0.02"/>
        </geometry>
        <material name="green"/>
        </visual>
    </link>

    <link name="br_l_ph" >
        <visual>
        <geometry>
            <box size="0.005 0.02 0.02"/>
        </geometry>
        <material name="blue"/>
        </visual>
    </link>
    <link name="br_r_ph" >
        <visual>
        <geometry>
            <box size="0.005 0.02 0.02"/>
        </geometry>
        <material name="green"/>
        </visual>
    </link>

    <link name="bl_l_ph" >
        <visual>
        <geometry>
            <box size="0.005 0.02 0.02"/>
        </geometry>
        <material name="blue"/>
        </visual>
    </link>
    <link name="bl_r_ph" >
        <visual>
        <geometry>
            <box size="0.005 0.02 0.02"/>
        </geometry>
        <material name="green"/>
        </visual>
    </link>    
  
<!-- pinhole joints -->
    <joint name ="fl_to_l_ph" type="fixed">
        <parent link="fl_fy" />
        <child link="fl_l_ph"/>
        <origin xyz="0 0 0" rpy="0 0.785 0 "/>
    </joint>
    <joint name ="fl_to_r_ph" type="fixed">
        <parent link="fl_fy" />
        <child link="fl_r_ph"/>
        <origin xyz="0 0 0" rpy="0 -0.785 0 "/>
    </joint>

    <joint name ="fr_to_l_ph" type="fixed">
        <parent link="fr_fy" />
        <child link="fr_l_ph"/>
        <origin xyz="0 0 0" rpy="0 0.785 0"/>
    </joint>
    <joint name ="fr_to_r_ph" type="fixed">
        <parent link="fr_fy" />
        <child link="fr_r_ph"/>
        <origin xyz="0 0 0" rpy="0 -0.785 0"/>
    </joint>

    <joint name ="br_to_l_ph" type="fixed">
        <parent link="br_fy" />
        <child link="br_l_ph"/>
        <origin xyz="0 0 0" rpy="0 0.785 0"/>
    </joint>
    <joint name ="br_to_r_ph" type="fixed">
        <parent link="br_fy" />
        <child link="br_r_ph"/>
        <origin xyz="0 0 0" rpy="0 -0.785 0"/>
    </joint>
    <joint name ="bl_to_l_ph" type="fixed">
        <parent link="bl_fy" />
        <child link="bl_l_ph"/>
        <origin xyz="0 0 0" rpy="0 0.785 0"/>
    </joint>
    <joint name ="bl_to_r_ph" type="fixed">
        <parent link="bl_fy" />
        <child link="bl_r_ph"/>
        <origin xyz="0 0 0" rpy="0 -0.785 0"/>
    </joint>        

</robot>